python_flask/app/__init__.py

```diff
@@ -9,8 +9,23 @@
 def hello():
     return "Hello world"
 
-@app.route('/api/clients')
+@app.route('/api/clients', methods = ['GET', 'POST'])
 def clients():
+    if request.method == 'POST':
```

Comment:
This is really routing logic, so it should be the responsibility of the router to switch on the HTTP verb. Also, you have one function doing the work of two routes, but there isn't really any overlap. I'd split this into two separate functions/routes. One for GET and one for POST.

BTW, I think this exposes a weakness in the test suite, because POSTing a new client should only return the created client, not the entire collection.